Add mention of new IERS server as a workaround (#9863)
astropy/astropy: docs/known_issues.rst

file. The root cause of this issue is that pytest now tries to pick up the
top-level ``test()`` function as a test, so we need to make sure that we set a
``test.__test__`` attribute on the function to ``False``.

> Workarounds for Unspported Versions

Reviewer: ```suggestion
Workarounds for Unsupported Versions
```

===================================

These are problems in *earlier* versions of Astropy which are caused by
changes to something outside of the control of Astropy itself.  Workarounds

> are provided as a way to potentially keep code that depends on older versions

Reviewer 2: ```suggestion
are provided as a way to potentially keep code that depends on the functioning
```

> functioning.  This are to be considered "as-is" or "best effort", though, as

Reviewer 2: ```suggestion
of older versions.  These are to be considered "as is" or "best effort," though, as
```

Reviewer 3: I think these two lines were meant as "keep (code that depends on older versions of astropy) functioning", so it should be rephrased to "support code that depends on older versions of astropy"  (using "support" as a replacement for "keep ... functioning")

> older versions of Astropy cannot be supported for arbitrarily long amounts of

Reviewer 2: ```suggestion
older versions of Astropy cannot be supported for arbitrarily long periods of
```

time.

IERS server changes
-------------------

In late 2019, the canonical server for providing the most precise earth rotation
parameters (the USNO IERS server) was taken offline for an extended period.
While current versions of Astropy have been updated to use a mirror server,

> older versions (i.e. prior to 3.2.3) still have as their default the server

Reviewer: ```suggestion
older versions (i.e., prior to 3.2.3) still have as their default the server
```

Reviewer 3: I'm :+1: on this change but it is, afaik, a style choice - do we have any English style guidelines anywhere that we have to follow?  If not, I recommend that we do _not_ create them...

Reviewer: We do have a guideline, thanks to @lglattly ! https://docs.astropy.org/en/latest/development/style-guide.html#abbreviations

> that is unavilable (at the time of this writing).  A workaround is to use the

Reviewer 3: ```suggestion
that is unavailable (at the time of this writing).  A workaround is to use the
```

following code::

    from astropy.utils import iers

>     iers.Conf.iers_auto_url.set('ftp://cddis.gsfc.nasa.gov/pub/products/iers/finals2000A.all')

Reviewer: I think this is more consistent with https://docs.astropy.org/en/stable/config/#changing-values-at-runtime 

```suggestion
    iers.conf.iers_auto_url = 'ftp://cddis.gsfc.nasa.gov/pub/products/iers/finals2000A.all'
```

> this must be executed *before* any code that makes use of IERS (much of

Reviewer 2: ```suggestion
This must be executed *before* any code that makes use of IERS (much of
```

> `astropy.coordinates` and significant portions of `astropy.time`) - the safest

Reviewer 2: ```suggestion
`astropy.coordinates` and significant portions of `astropy.time`) — the safest
```

> choice is to run it before *any* other Astropy code.

Reviewer 3: "the safest choice" is kind of scary catch-all language.  Can we be more specific here?  e.g., "This code must be executed before any other code that uses.... If `coordinates` or `time` features are used prior to invoking the above lines, the original URL will be cached, and the cache cannot (easily?) be cleared."

Or, if the cache can be cleared, link to those docs.  

I don't mean to draw this out, as this text is already helpful, but "safest" suggests other options are unsafe, and frankly, I want to know what the underlying operations are... I think you could also do something like clear cache, then reset the URL, then start doing stuff?
